src/settings/LoanHistory/LoanHistory.js

```diff
@@ -58,25 +60,28 @@ class LoanHistorySettings extends React.Component {
       config = defaultConfig;
     }
 
-    return config.closingType !== closingTypesMap.INTERVAL ? defaultConfig : config;
+    return config;
   }
 
   validate = values => {
     const errors = {};
-    const intervalValue = Number(values.intervalValue);
-    const isNumberValid = isInteger(intervalValue) && intervalValue > 0;
-    const isIntervalValueValid = !isNumberValid && (values.closingType === closingTypesMap.INTERVAL || !isEmpty(values.intervalValue));
 
-    if (isIntervalValueValid) {
-      errors.intervalValue = { _error: <FormattedMessage id="ui-circulation.settings.loanHistory.validate.intervalValue" /> };
-    }
+    Object.values(closedLoansRules).forEach(item => {
+      const durationValue = values[item] && Number(values[item].duration);
+      const isNumberValid = isInteger(durationValue) && durationValue > 0;
 
-    const isIntervalTypeValid = !values.intervalType && values.closingType === closingTypesMap.INTERVAL;
+      if (values.closingType[item] === closingTypesMap.INTERVAL && !isNumberValid) {
+        const duration = { _error: <FormattedMessage id="ui-circulation.settings.loanHistory.validate.intervalValue" /> };
+        setWith(errors, [item, 'duration'], duration);
+      }
 
-    if (isIntervalTypeValid) {
-      errors.intervalType = { _error: <FormattedMessage id="ui-circulation.settings.loanHistory.validate.selectContinue" /> };
-    }
+      const isIntervalIdValid = values[item] && values[item].intervalId;
 
+      if (values.closingType[item] === closingTypesMap.INTERVAL && !isIntervalIdValid) {
+        const intervalId = { _error: <FormattedMessage id="ui-circulation.settings.loanHistory.validate.selectContinue" /> };
+        setWith(errors, [item, 'intervalId'], intervalId);
+      }
+    });
     return errors;
```

Comment:
Add an empty line before return. Please go through previous my comments I made in other your PR in order to avoid repeating ones. Check other places in the PR.